src/app/beer_garden/db/mongo/models.py

```diff
@@ -773,3 +776,53 @@ class FileChunk(MongoModel, Document):
 
 class RawFile(Document):
     file = FileField()
+
+
+class Role(Document):
+    name = StringField()
+    description = StringField()
+    permissions = ListField(field=StringField(), validation=validate_permissions)
+
+    meta = {
+        "indexes": [{"name": "unique_index", "fields": ["name"], "unique": True}],
+    }
+
+
+class RoleAssignment(EmbeddedDocument):
+    domain = StringField()
+    role = ReferenceField("Role")
+
+
+class User(Document):
+    username = StringField(required=True)
+    password = StringField()
+    role_assignments = EmbeddedDocumentListField("RoleAssignment")
+
+    meta = {
+        "indexes": [{"name": "unique_index", "fields": ["username"], "unique": True}],
+    }
+
+    def set_password(self, password: str):
+        """This helper should be used to set the user's password, rather than directly
+        assigning a value. This ensures that the password is stored as a hash rather
+        than in plain text
+
+        Args:
+            password: String to set as the user's password.
+
+        Returns:
+            None
+        """
+        self.password = custom_app_context.hash(password)
```

Comment:
This was what the existing code used to has the passwords and it seemed reasonably / easy enough so I just went with it rather than exploring other options.  If anyone is more familiar with this (or password hashing in general) and has feelings on it, please do share.

Reply:
Yes, I agree that we should take a more in depth look at this particular library and examine other options if necessary.  I purposefully wrote `set_password`, `verify_password` and the corresponding tests such that the particular library and hashing algorithm used should be easily swappable without having to change much else.